cpp/src/arrow/compute/kernels/scalar_temporal_test.cc

```diff
@@ -1324,6 +1357,592 @@ TEST_F(ScalarTemporalTest, TestTemporalDifferenceZoned) {
+  CheckScalarUnary(op, unit, times, unit, round_1_seconds, &round_to_1_seconds);
+  CheckScalarUnary(op, unit, times, unit, round_1_minutes, &round_to_1_minutes);
+  CheckScalarUnary(op, unit, times, unit, round_1_hours, &round_to_1_hours);
+  CheckScalarUnary(op, unit, times, unit, round_1_days, &round_to_1_days);
+  CheckScalarUnary(op, unit, times, unit, round_1_weeks, &round_to_1_weeks);
+  CheckScalarUnary(op, unit, times, unit, round_1_months, &round_to_1_months);
+  CheckScalarUnary(op, unit, times, unit, round_1_quarters, &round_to_1_quarters);
+  CheckScalarUnary(op, unit, times, unit, round_1_years, &round_to_1_years);
+
+  CheckScalarUnary(op, unit, times, unit, round_15_nanoseconds, &round_to_15_nanoseconds);
+  CheckScalarUnary(op, unit, times, unit, round_15_microseconds,
+                   &round_to_15_microseconds);
+  CheckScalarUnary(op, unit, times, unit, round_15_milliseconds,
+                   &round_to_15_milliseconds);
+  CheckScalarUnary(op, unit, times, unit, round_15_seconds, &round_to_15_seconds);
+  CheckScalarUnary(op, unit, times, unit, round_15_minutes, &round_to_15_minutes);
+  CheckScalarUnary(op, unit, times, unit, round_15_hours, &round_to_15_hours);
+  CheckScalarUnary(op, unit, times, unit, round_15_days, &round_to_15_days);
+  CheckScalarUnary(op, unit, times, unit, round_15_weeks, &round_to_15_weeks);
+  CheckScalarUnary(op, unit, times, unit, round_15_months, &round_to_15_months);
+  CheckScalarUnary(op, unit, times, unit, round_15_quarters, &round_to_15_quarters);
+  CheckScalarUnary(op, unit, times, unit, round_15_years, &round_to_15_years);
+}
+
+TEST_F(ScalarTemporalTest, TestCeilFloorRoundTemporalBrussels) {
+  RoundTemporalOptions round_to_1_hours = RoundTemporalOptions(1, CalendarUnit::HOUR);
+  RoundTemporalOptions round_to_2_hours = RoundTemporalOptions(2, CalendarUnit::HOUR);
+  auto unit = timestamp(TimeUnit::NANO, "Europe/Brussels");
+
+  const char* times = R"(["2021-12-23 12:17:00", null])";
+  const char* ceil_1_hours = R"(["2021-12-23 13:00", null])";
+  const char* ceil_2_hours = R"(["2021-12-23 13:00", null])";
+  const char* floor_1_hours = R"(["2021-12-23 12:00", null])";
+  const char* floor_2_hours = R"(["2021-12-23 11:00", null])";
+  const char* round_1_hours = R"(["2021-12-23 12:00", null])";
+  const char* round_2_hours = R"(["2021-12-23 13:00", null])";
+
+  CheckScalarUnary("ceil_temporal", unit, times, unit, ceil_1_hours, &round_to_1_hours);
+  CheckScalarUnary("ceil_temporal", unit, times, unit, ceil_2_hours, &round_to_2_hours);
+  CheckScalarUnary("floor_temporal", unit, times, unit, floor_1_hours, &round_to_1_hours);
+  CheckScalarUnary("floor_temporal", unit, times, unit, floor_2_hours, &round_to_2_hours);
+  CheckScalarUnary("round_temporal", unit, times, unit, round_1_hours, &round_to_1_hours);
+  CheckScalarUnary("round_temporal", unit, times, unit, round_2_hours, &round_to_2_hours);
+}
+
+TEST_F(ScalarTemporalTest, TestCeilFloorRoundTemporalKolkata) {
```

Comment:
nit: for context, can we note that this test is here to test the a case where `date` (and hence our kernels) uses historical TZ info while Pandas (and possibly other libraries) do not?

Reply:
Otherwise I fear confusion when someone in the future looks and wonders why Kolkata was an especially important time zone to test :)

Reply:
Yeah, that makes sense.

Reply:
[Comment added](https://github.com/apache/arrow/blob/e8f9a621e37fe989e7ae5297886ab7decc069203/cpp/src/arrow/compute/kernels/scalar_temporal_test.cc#L1912-L1948).

It seems reality is [more nuanced](https://www.timeanddate.com/time/zone/india/kolkata) than [Wikipedia states](https://en.wikipedia.org/wiki/Calcutta_Time).